### ISSUE TITLE
[arcface torch] Loss and learning rate

### ISSUE THREAD

**Issue by jacqueline-weng** [deepinsight/insightface#1974]
As it is designed, learning rate raises up linearly in the warmup stage and decreases exponentially in training stage.
During warmup stage, the loss would first decrease before it increases with the growing of learning rate.
The optimal loss pattern is that it will reach minima just at the time when learning rate finishes warming up.
However, recently I found the loss bouncing back so quick when I transferred the training from a 8gpu machine to a 4gpu machine (the sum cuda memory is approximately same).
Is this unexpected loss pattern related to number of GPUs?
Are there any parameters I could adjust?
![image](https://user-images.githubusercontent.com/62649623/163515103-1fe3bf5b-6571-4953-bf0e-f2bafe1192f6.png)

**Comment by cao-nv** [comment]
For most cases in my experience, you should shrink the learning rate as batch size reduced.

**Comment by jacqueline-weng** [comment]
> 

Do you mean shrinking number of total step to make lr changes more quickly? (will it equal with decreasing the number of training epoch?)

**Comment by jacqueline-weng** [comment]
> For most cases in my experience, you should shrink the learning rate as batch size reduced.

Do you mean shrinking number of total step to make lr changes more quickly? (Is it equivalent with decreasing the number of training epoch?)

**Comment by cao-nv** [comment]
> > For most cases in my experience, you should shrink the learning rate as batch size reduced.
> 
> Do you mean shrinking number of total step to make lr changes more quickly? (Is it equivalent with decreasing the number of training epoch?)

I mean you should reduce the learning rate value in the config file (i.e maximum learning rate). 
When you use a config for 8-GPUs on a 4-GPUs machine, the actual batch-size changes from `B x 8` to `B x 4` (`B` is batch-size value in the config file). Thus you should reduce the learning rate by approximately a half.
